### ISSUE TITLE
[FEA] Add Functionality to index dataset partitions

### ISSUE THREAD

**Issue by jperez999** [NVIDIA/NVTabular#170]
In order to comply with functionality of pytorch dataloader, the NVTabular dataset, during iteration, should be able to take in start and stop indexes and it should be able to randomize partition order access via kwarg shuffle. Something like the following:
```
nvt.Dataset().to_iter(start=0, end=10, shuffle=True)
-shuffle partitions
-return generator for the specified indexes

**Comment by jperez999** [comment]
fyi @rjzamora

**Comment by jperez999** [comment]
EDIT: instead of handling shuffling of partitions in teh dataset. the dataloader worker logic will handle the partition access. So api should look like:
nvt.Dataset().to_iter(indexes=[list of indexes to access])
- iterate over list indexes and access those.

**Comment by jperez999** [comment]
if no indexes given traverse the whole thing.

**Comment by rjzamora** [comment]
Sounds good - The DataFrameIter class would look something like:

```python
class DataFrameIter:
    def __init__(self, ddf, columns=None, indices=None):
        self._ddf = ddf
        self.columns = columns
        self.indices=indices

    def __len__(self):
        return self._ddf.npartitions

    def __iter__(self):
        indices = self.indices or list(range(self._ddf.npartitions))
        for i in indices:
            part self._ddf.get_partition(i):
            if self.columns:
                yield part[self.columns].compute(scheduler="synchronous")
            else:
                yield part.compute(scheduler="synchronous")
            part = None
```

and you just need to plumb it through to the `Dataset.to_iter` method.  Is this something you want to tackle @jperez999 ?

**Comment by jperez999** [comment]
yeah I will take this.
